include/hybridmap/NeutralAtomArchitecture.hpp

```diff
@@ -0,0 +1,485 @@
+    [[nodiscard]] std::uint16_t getNAodCoordinates() const {
+      return nAodCoordinates;
+    }
+    [[nodiscard]] std::uint16_t getNAodIntermediateLevels() const {
+      return nAodIntermediateLevels;
+    }
+    [[nodiscard]] fp getInterQubitDistance() const {
+      return interQubitDistance;
+    }
+    [[nodiscard]] fp getInteractionRadius() const { return interactionRadius; }
+    [[nodiscard]] fp getBlockingFactor() const { return blockingFactor; }
+  };
+
+  /**
+   * @brief Class to store the parameters of a neutral atom architecture
+   * @details
+   * The parameters of a neutral atom architecture are:
+   * - number of qubits
+   * - gate times
+   * - gate average fidelities
+   * - shuttling times
+   * - shuttling average fidelities
+   * - decoherence times
+   * The parameters are loaded from a JSON file.
+   * The difference to the properties is that the parameters can change
+   * from run to run.
+   */
+  struct Parameters {
+    /**
+     * @brief Class to store the decoherence times of a neutral atom
+     * architecture
+     * @details
+     * The decoherence times of a neutral atom architecture are:
+     * - T1
+     * - T2
+     * - effective decoherence time
+     */
+    class DecoherenceTimes {
+    protected:
+      fp                  tEff;
+      [[maybe_unused]] fp t1;
+      [[maybe_unused]] fp t2;
+
+    public:
+      DecoherenceTimes() = default;
+      DecoherenceTimes(fp t1, fp t2)
+          : tEff(t1 * t2 / (t1 + t2)), t1(t1), t2(t2) {}
+      [[nodiscard]] fp getTEff() const { return tEff; }
+    };
+    CoordIndex                nQubits;
+    std::map<std::string, fp> gateTimes;
+    std::map<std::string, fp> gateAverageFidelities;
+    std::map<OpType, fp>      shuttlingTimes;
+    std::map<OpType, fp>      shuttlingAverageFidelities;
+    DecoherenceTimes          decoherenceTimes;
+  };
+
+protected:
+  Properties properties{};
+  Parameters parameters;
+
+  std::vector<Coordinate>           coordinates;
+  SymmetricMatrix                   swapDistances;
+  std::vector<std::set<CoordIndex>> nearbyCoordinates;
+
+  /**
+   * @brief Create the coordinates.
+   */
+  void createCoordinates();
+  /**
+   * @brief Compute the swap distances between the coordinates
+   * @details
+   * The swap distances are computed using the coordinates of the qubits.
+   * The swap distance is the distance between the qubits in terms of
+   * edges in the resulting connectivity graph. This can be computed
+   * beforehand.
+   */
+  void computeSwapDistances(fp interactionRadius);
+  /**
+   * @brief Compute the nearby coordinates for each coordinate
+   * @details
+   * The nearby qubits are the qubits that are close enough to be connected
+   * by an edge in the resulting connectivity graph. This can be be computed
+   * beforehand.
+   */
+  void computeNearbyCoordinates();
+
+public:
+  std::string name;
+
+  // Constructors
+  NeutralAtomArchitecture() = delete;
```

Comment:
As an architecture without the json doesn't really make sense I thought I might prevent it. But can also allow it obviously.

Reply:
Given how C++ sometimes works, it's probably better to make the architecture default constructible. I see practically no downside and a slight plus in the flexibility. In the future there might be even more constructors that do not require a file/filesystem access at all (but, e.g., use QDMI to populate the individual properties). There it might be nice to first default construct and then populate.

Reply:
Added back the default constructor 👍